code/src/nuvla/connector/kubernetes.py

```diff
@@ -169,15 +169,84 @@ def list(self, filters=None):
         pass
 
+    def _get_container_logs(self, namespace, values, since_opt, lines: int) -> str:
+        tail_lines=lines # wtf
+        tail_lines=5 # wtf
+        logs_string=str()
+        pod_unique_id = str(values["metadata"]["name"])
+        log.debug('Unique pod ID: %s ', pod_unique_id)
+        try:
+            for containers_list in values['spec']['containers']:
+                container = str(containers_list['name'])
+                log.debug('Found container: %s ', container)
+                list_opts_log = ['--timestamps=true', '--tail', str(tail_lines),
+                                 '--namespace', namespace] + since_opt
+                container_opts = \
+                ['pod/' + pod_unique_id, '--container=' + container]
+                cmd = \
+                self.build_cmd_line(['logs'] + container_opts + list_opts_log)
+                log.debug('Generated logs command line : %s', cmd)
+                header_line = "\n\nLog last " + str(tail_lines) + \
+                " lines for Container " + \
+                container + " in Pod " + pod_unique_id + " \n"
+                log.debug('Header line : %s', header_line)
+                logs_string = \
+                logs_string + header_line + execute_cmd(cmd).stdout
+                log.debug('_get_container_logs logs string : %s', logs_string)
+        except Exception as e_cont:
+            ex_string = "There was a problem getting logs from Pod " \
+            + pod_unique_id
+            log.debug('An exception caught: %s ', e_cont)
+            logs_string = str(ex_string)
+        return logs_string 
+
+    def _get_the_pods(self, namespace, values, since_opt, lines: int) -> str:
+        logs_string = str()
+        log.debug('Starting _get_containers_logs.')
+        for items_list in values['items']:
+            if str(items_list["kind"]) == 'Pod':
+                logs_string = logs_string + \
+                self._get_container_logs(namespace, items_list, since_opt, lines)
+        log.debug('_get_containers FINAL log string : %s', logs_string)
+
+        return logs_string
+
+    def _get_the_logs(self, namespace, since_opt, lines: int) -> str:
+        list_opts_pods = ['-o', 'json', '--namespace', namespace]
+        cmd_pods = self.build_cmd_line(['get', 'pods'] + list_opts_pods)
+        log.debug('Generated command line to get pods: %s', cmd_pods)
+        try:
+            all_json_out = json.loads(execute_cmd(cmd_pods).stdout)
+            try:
+                log.info('Getting containers...')
+                logs_string = \
+                self._get_the_pods(namespace, all_json_out, since_opt, lines)
+            except Exception as e_cont:
+                log.debug('Fetching Containers failed: %s ', e_cont)
+        except Exception as e_json:
+            log.debug('Fetching JSON failed: %s', e_json)
+        # FIXME I leave for now
+        log.info('Container logs string: %s', logs_string)
+        return logs_string
```

Comment:
`logs_string` is not defined in the method's top-level scope. In case of any exception in the above `try/except`-s, `logs_string` will not be defined and when reaching line 230 the interpreter will fail with `variable not defined`.

Reply:
Now I have a default definition for `logs_string`.
This is a very basic default message with the correct timestamp at the start.
This is needed as if there is not the correct timestamp, then the string will fail the test in Clojure template and we get an endlessly turning loading circle on the UI window.